frontend/src/app/core/services/errors.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { IssueMessage } from '@shared/models/issues/issue-message';
+import { HttpInternalService } from '@core/services/http-internal.service';
+import { StackTrace } from '@shared/models/issues/stack-trace';
+import { IssueEnvironment } from '@shared/models/issues/issue-environment';
+import { HttpResponseErrorMessage } from '@shared/models/issues/http-response.message';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { environment } from '@env/environment';
+import * as stackTraceParser from 'stacktrace-parser';
+import { ToastNotificationService } from './toast-notification.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ErrorsService implements OnDestroy {
+    private unsubscribe$: Subject<void> = new Subject<void>();
+
+    constructor(private httpService: HttpInternalService, private toastNotification: ToastNotificationService) { }
+
+    log(error: any) {
+        const issueMessage = this.addContextInfo(error);
+        console.log(issueMessage);
+        this.httpService.postFullRequest(`${environment.collectorUrl}/issues`, issueMessage)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe(() => {
+                this.toastNotification.info('Sent error to collector.', null, 1700);
+            }, () => {
+                this.toastNotification.info('Are you sure that collector is running?', null, 1700);
+            });
+    }
+
+    private addContextInfo(error: any): IssueMessage {
+        return {
+            occurredOn: new Date(),
+            issueDetails: {
+                url: window.location.href,
+                errorMessage: error.message === '' ? 'Script error' : error.message,
```

Comment:
You can just use error.message instead of error.message === ''

Reply:
i'm not sure that _error.message === ''_ is same to just _error.message_. in such way error name is empty
![5](https://user-images.githubusercontent.com/66973922/128915360-7724b051-177b-49ea-9c6b-699c94a32af1.png)

Reply:
Okay